src/initialization/FGLinearization.cpp

```diff
@@ -20,23 +20,19 @@
 #include "FGLinearization.h"
 #include <ctime>
 
-namespace JSBSim {
 
-// TODO make FGLinearization have X,U,Y selectable by xml config file
+namespace JSBSim {
 
-FGLinearization::FGLinearization(FGFDMExec * fdm, int mode)
+FGLinearization::FGLinearization(FGFDMExec * fdm) : aircraft_name(fdm->GetAircraft()->GetAircraftName())
 {
-    std::cout << "\nlinearization: " << std::endl;
-    std::clock_t time_start=clock(), time_linDone;
-    FGStateSpace ss(fdm);
-
+    auto ss = FGStateSpace(fdm);
     ss.x.add(new FGStateSpace::Vt);
     ss.x.add(new FGStateSpace::Alpha);
     ss.x.add(new FGStateSpace::Theta);
     ss.x.add(new FGStateSpace::Q);
 
     // get propulsion pointer to determine type/ etc.
-    FGEngine * engine0 = fdm->GetPropulsion()->GetEngine(0);
+    auto engine0 = fdm->GetPropulsion()->GetEngine(0);
```

Comment:
What for ? Current code doesn't hurt: I suggest to keep the existing code.
```suggestion
    FGEngine * engine0 = fdm->GetPropulsion()->GetEngine(0);
```

Reply:
I changed that because the `GetEngine()` function has changed since this code was last compiled. I think it now returns a smart pointer. Changing it to `auto` was a fast fix when I was trying to get the code to compile. I should've changed it to the new type afterwards though, but forgot.

Reply:
Yes, indeed. I forgot that the list of engines in `FGPropulsion` has moved to smart pointers. No problem in using `auto` in that particular case.